extensions/mssql/src/errorDiagnostics/errorDiagnosticsProvider.ts

```diff
@@ -62,6 +62,13 @@ export class ErrorDiagnosticsProvider extends SqlOpsFeature<any> {
 						logDebug(`ErrorDiagnosticsProvider: Error Code ${errorInfo.errorCode} indicates certificate validation has failed, launching error dialog with instructionText.`);
 						return await this.showCertValidationDialog(restoredProfile, errorInfo.errorMessage, errorInfo.messageDetails);
 					}
+					else if (errorInfo.errorCode === ErrorDiagnosticsConstants.MssqlLoginFailedForUserErrorCode
+						&& (connection.serverName.endsWith(ErrorDiagnosticsConstants.AzureSQLSuffixGlobal) || connection.serverName.endsWith(ErrorDiagnosticsConstants.AzureSQLSuffixUSNational))) {
+						return await this.showDiagnoseErrorDialog(restoredProfile, errorInfo.errorCode, errorInfo.errorMessage, errorInfo.messageDetails);
```

Comment:
Any reason to have this condition? Diagnostic RP doesn't seem to come anytime soon so I think we can have a single case for this dialog.